packages/flutter_tools/lib/src/commands/upgrade.dart

```diff
@@ -132,11 +132,12 @@ class UpgradeCommandRunner {
     } else {
       tag = 'v${gitTagVersion.x}.${gitTagVersion.y}.${gitTagVersion.z}';
     }
-    final RunResult runResult = await runCheckedAsync(<String>[
-      'git', 'reset', '--hard', tag,
-    ], workingDirectory: Cache.flutterRoot);
-    if (runResult.exitCode != 0) {
-      throwToolExit('Failed to restore branch from hotfix.');
+    try {
+      await runCheckedAsync(<String>[
+        'git', 'reset', '--hard', tag,
+      ], workingDirectory: Cache.flutterRoot);
+    } catch (error) {
+      throwToolExit('git reset failed: $error.');
```

Comment:
For `throwToolExit` it would be good to explain what the tool was doing and a hint about what the user should do to get unblocked.

Reply:
Done.